woocommerce/woocommerce-ios
# [IPPInAppFeedback] Banner dismiss logic

Closes: #8589
Part of https://github.com/woocommerce/woocommerce-ios/issues/8530

## Description

This PR adds the logic to dismiss the IPP feedback banner.

## TODO

- [x] Dismisses the IPP banner upon tapping
- [x] Presents confirmation screen upon dismissal
- [x] Dismisses the IPP banner upon survey completion.

## Changes
- Updates the `setFeatureAnnouncementDismissed` AppSettingsStore/AppSettingsAction to accept a new `remindAfter` parameter, this allows us to specify a number of days for the reminder to be triggered (previous default to 14)
- Adds an `UIAlertController` to be shown when the IPP banner is dismissed, with 2 buttons: "Remind me later", and "Don't show again".
- Once the IPP banner is dismissed, we update the feedback status to `.dismissed`, and the feedback visibility to be reminded later or never.
- Updated, and added Unit Tests


## Testing instructions
- Go to My Store > Orders and see the IPP feedback banner appear
- Tap on "x" and see a confirmation alert with 2 options: "remind me later", and "Don't show again"
- Confirm that the banner dismisses no matter the option you tap
- Note: If Order Creation feedback banner should appear in your store, this will show after we've dismissed the IPP banner.

![Simulator Screen Recording - iPhone 11 Pro - 2023-01-19 at 10 19 40](https://user-images.githubusercontent.com/3812076/213340206-f20a6397-e0c5-473a-8eeb-dde96bd53b7c.gif)

## Screenshots



---
